src/editors/containers/ProblemEditor/components/EditProblemView/hooks.test.js

```diff
@@ -154,88 +200,134 @@ describe('EditProblemView hooks parseState', () => {
       it('returns true for single select problem with empty title', () => {
         window.tinymce.editors = { 'answer-A': { getContent: () => '' }, 'answer-B': { getContent: () => 'sOmevALUe' } };
         const expected = hooks.checkForNoAnswers({
-          openNoAnswerModal,
+          openSaveWarningModal,
           problem: {
             ...problem,
             answers: [{ id: 'A', title: '', correct: true }, { id: 'B', title: 'sOmevALUe', correct: false }],
           },
         });
-        expect(openNoAnswerModal).toHaveBeenCalled();
+        expect(openSaveWarningModal).toHaveBeenCalled();
         expect(expected).toEqual(true);
       });
       it('returns true for single select with title but no correct answer', () => {
         window.tinymce.editors = { 'answer-A': { getContent: () => 'sOmevALUe' } };
         const expected = hooks.checkForNoAnswers({
-          openNoAnswerModal,
+          openSaveWarningModal,
           problem: {
             ...problem,
             answers: [{ id: 'A', title: 'sOmevALUe', correct: false }, { id: 'B', title: '', correct: false }],
           },
         });
-        expect(openNoAnswerModal).toHaveBeenCalled();
+        expect(openSaveWarningModal).toHaveBeenCalled();
         expect(expected).toEqual(true);
       });
       it('returns true for single select with title and correct answer', () => {
         window.tinymce.editors = { 'answer-A': { getContent: () => 'sOmevALUe' } };
         const expected = hooks.checkForNoAnswers({
-          openNoAnswerModal,
+          openSaveWarningModal,
           problem: {
             ...problem,
             answers: [{ id: 'A', title: 'sOmevALUe', correct: true }],
           },
         });
-        expect(openNoAnswerModal).not.toHaveBeenCalled();
+        expect(openSaveWarningModal).not.toHaveBeenCalled();
         expect(expected).toEqual(false);
       });
     });
   });
+  describe('checkForSettingDiscrepancy', () => {
+    const openSaveWarningModal = jest.fn();
+    const problem = problemState;
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+    it('returns true for setting discrepancies', () => {
+      jest.mock('../../data/ReactStateSettingsParser', () => (
+        jest.fn().mockImplementationOnce(() => ({
+          getSettings: () => mockGetSettings,
+          parseRawOlxSettings: () => mockParseRawOlxSettingsDiscrepancy,
+        }))
+      ));
+      const mockRawOLXWithSettings = '<problem show_reset_button="true">rawOLX</problem>';
+      const refMockWithSettings = { current: { state: { doc: { toString: () => mockRawOLXWithSettings } } } };
+      const expected = hooks.checkForSettingDiscrepancy({
+        openSaveWarningModal,
+        problem,
+        ref: refMockWithSettings,
+      });
+      expect(openSaveWarningModal).toHaveBeenCalled();
+      expect(expected).toEqual(true);
+    });
+    it('returns false for setting discrepancies', () => {
```

Comment:
```suggestion
    it('returns false when there are no setting discrepancies', () => {
```

- I assume that is what you are testing here